apps/web/components/apps/AppPage.tsx

```diff
@@ -70,22 +73,63 @@ export const AppPage = ({
   dependencies,
   concurrentMeetings,
   paid,
-  dirName,
 }: AppPageProps) => {
   const { t, i18n } = useLocale();
+  const router = useRouter();
   const searchParams = useCompatSearchParams();
 
   const hasDescriptionItems = descriptionItems && descriptionItems.length > 0;
 
-  const mutation = useAddAppMutation(null, {
-    onSuccess: (data) => {
-      if (data?.setupPending) return;
-      showToast(t("app_successfully_installed"), "success");
-    },
-    onError: (error) => {
-      if (error instanceof Error) showToast(error.message || t("app_could_not_be_installed"), "error");
-    },
-  });
+  const mutation = useAddAppMutation(null);
+
+  const [isLoading, setIsLoading] = useState<boolean>(mutation.isPending);
+
+  const handleAppInstall = () => {
+    setIsLoading(true);
+    if (isConferencing(categories)) {
+      mutation.mutate(
+        {
+          isOmniInstall: true,
+          type,
+          variant,
+          slug,
+          returnTo:
+            WEBAPP_URL +
+            getAppOnboardingUrl({
+              slug,
+              step: AppOnboardingSteps.EVENT_TYPES_STEP,
+            }),
+        },
+        {
+          onSuccess: (data) => {
+            if (data?.setupPending) return;
+            // for non-oAuth apps
+            router.push(
+              getAppOnboardingUrl({
+                slug,
+                step: AppOnboardingSteps.EVENT_TYPES_STEP,
+              })
+            );
+            showToast(t("app_successfully_installed"), "success");
+          },
+          onError: (error) => {
+            if (error instanceof Error) showToast(error.message || t("app_could_not_be_installed"), "error");
+            setIsLoading(false);
+          },
+        }
+      );
+    } else if (
+      !doesAppSupportTeamInstall({
+        appCategories: categories,
+        concurrentMeetings: concurrentMeetings,
+        isPaid: !!paid,
+      })
+    ) {
+      mutation.mutate({ type });
+    } else {
+      router.push(getAppOnboardingUrl({ slug, step: AppOnboardingSteps.ACCOUNTS_STEP }));
```

Comment:
@SomayChauhan what apps are going to fall into this condition? I'm trying to test them locally but couldn't pinpoint any obvious ones.

Reply:
all analytics apps, CRM apps, payment apps, and a few for other category apps like - qrcode, basecamp and giphy